pages/_app.js

```diff
@@ -45,6 +45,20 @@ export default function App({ Component, pageProps }) {
     router.push(`/`);
   }
 
+  function handleDeleteBook(currentBook) {
+    setBooks(books.filter((book) => book.id !== currentBook.id));
+
+    router.push("/");
+  }
+
+  function handleDeleteClick(isClicked, currentBook) {
+    if (isClicked) {
+      handleDeleteBook(currentBook);
+    } else {
+      router.push("/");
+    }
+  }
```

Comment:
here you can simply refactor the router.push into the handleCancel function. Then hand over the handleDeleteBook to the onDelete prop of the modal and the handleCancel to the onCancel prop

Reply:
Changes:
```  function handleDeleteBook(currentBook) {
    setBooks(books.filter((book) => book.id !== currentBook.id));

    router.push("/");
  }
```
and
```
  function handleCancel() {
    router.push("/");
  }
´´´
to pass them to props.